castor-service/src/test/java/io/carbynestack/castor/service/persistence/cache/ReservationCachingServiceTest.java

```diff
@@ -58,282 +73,434 @@ public void setUp() {
+    UUID tupleChunkId = UUID.fromString("3fd7eaf7-cda3-4384-8d86-2c43450cbe63");
+    long requestedStartIndex = 42;
+    long requestedLength = 21;
+    ReservationElement re =
+        new ReservationElement(tupleChunkId, requestedLength, requestedStartIndex);
+    String reservationId = "testReservation";
+    TupleType tupleType = MULTIPLICATION_TRIPLE_GFP;
+    Reservation r = new Reservation(reservationId, tupleType, singletonList(re));
+    long existingFragmentStartIndex = 0;
+    long existingFragmentEndIndex = 99;
+    TupleChunkFragmentEntity existingFragment =
+        TupleChunkFragmentEntity.of(
+            tupleChunkId,
+            tupleType,
+            existingFragmentStartIndex,
+            existingFragmentEndIndex,
+            ActivationStatus.UNLOCKED,
+            null);
+
+    when(tupleChunkFragmentStorageServiceMock.splitAt(
+            existingFragment, requestedStartIndex + requestedLength))
+        .thenReturn(existingFragment);
+    when(tupleChunkFragmentStorageServiceMock.splitBefore(existingFragment, requestedStartIndex))
+        .thenReturn(existingFragment);
+    when(tupleChunkFragmentStorageServiceMock.findAvailableFragmentForChunkContainingIndex(
+            tupleChunkId, requestedStartIndex))
+        .thenReturn(Optional.of(existingFragment));
+
+    reservationCachingService.keepAndApplyReservation(r);
+    verify(tupleChunkFragmentStorageServiceMock, times(1)).update(existingFragment);
+    verify(tupleChunkFragmentStorageServiceMock, times(1))
+        .splitBefore(existingFragment, requestedStartIndex);
+    verify(tupleChunkFragmentStorageServiceMock, times(1))
+        .splitAt(existingFragment, requestedStartIndex + requestedLength);
+    verify(consumptionCachingServiceMock)
+        .keepConsumption(anyLong(), eq(tupleType), eq(requestedLength));
+  }
 
-    when(reservationElementMock.getTupleChunkId()).thenReturn(chunkId);
-    when(reservationElementMock.getStartIndex()).thenReturn(tupleStartIndex);
-    when(reservationElementMock.getReservedTuples()).thenReturn(requestedTuples);
-    when(reservationMock.getTupleType()).thenReturn(tupleType);
-    when(reservationMock.getReservations()).thenReturn(singletonList(reservationElementMock));
-    when(tupleChunkMetaDataStorageServiceMock.getTupleChunkData(chunkId)).thenReturn(metaDataMock);
-    when(metaDataMock.getStatus()).thenReturn(ActivationStatus.UNLOCKED);
-    when(metaDataMock.getTupleType()).thenReturn(tupleType);
-    when(metaDataMock.getReservedMarker()).thenReturn(tupleStartIndex);
-    when(metaDataMock.getNumberOfTuples()).thenReturn(totalTuplesInChunk);
+  @Test
+  public void
+      givenNoReservationReceivedInTime_whenGetReservationWithRetry_thenThrowCastorServiceException()
+          throws ExecutionException, InterruptedException, TimeoutException {
+    String reservationId = "testReservationId";
+    TupleType tupleType = MULTIPLICATION_TRIPLE_GFP;
```

Comment:
Used on only once in remainder of body -> redundant

Reply:
see discussion [above](https://github.com/carbynestack/castor/pull/29#discussion_r868054674)